### PR TITLE
Fix some issues with compositing rgba geotiffs

### LINKED ISSUE #912: Alpha on GDAL source still broken
See https://github.com/girder/large_image/pull/909#issuecomment-1205797033

```py
path = '/Users/bane/Desktop/rasters/Django-Large-Image/4-ortho-rgb-cog.tif'

src = large_image.open(path, projection='EPSG:3857', style={
 'bands': [
    {'band': 1, 'palette': 'R'},
    {'band': 2, 'palette': 'G'},
    {'band': 3, 'palette': 'B'}
  ]
})
thumb = src.getThumbnail(encoding='PNG')[0]
thumb
```
download

but:

```py
path = '/Users/bane/Desktop/rasters/Django-Large-Image/4-ortho-rgb-cog.tif'

src = large_image.open(path, projection='EPSG:3857')
thumb = src.getThumbnail(encoding='PNG')[0]
thumb
```
download

### PR DESCRIPTION
**Pull request by manthey**
Closes #912.

This implements some tests to verify correct behavior.

### CODE REVIEW

**Comment by manthey** [comment]
I create a RGBA geotiff with some synthetic data to use as a test.

File: test/test_source_gdal.py [hunk 1]
```diff
@@ -580,3 +580,24 @@ def testNoData():
...
+        imagePath, projection='EPSG:3857',
+        style={'bands': [
+            {'band': 1, 'palette': 'R'},
+            {'band': 2, 'palette': 'G'},
+            {'band': 3, 'palette': 'B'}]})
+    assert source.getThumbnail(encoding='PNG')[0] == base
+    assert not (source.getThumbnail(format='numpy')[0] - basenp).any()
+    source = large_image_source_gdal.open(
+        imagePath)
+    assert numpy.count_nonzero(source.getThumbnail(format='numpy')[0][:, :, 3] == 255) > 30000
```
Comment (banesullivan) [inline test/test_source_gdal.py:603]: Awesome, thanks for adding this test. Was the bug in #912 repeatable with this test file, `'rgba_geotiff.tiff'`?

Comment (banesullivan) [reply test/test_source_gdal.py:603]: Ah just saw your comment that you created that file for this test. Perfect!

**Review by banesullivan — APPROVED** [review]

File: test/test_source_gdal.py (hunk 1 above)
Comment (manthey) [reply test/test_source_gdal.py:603]: Yes.

**Pull request merged**